.github/workflows/publish_wheels.yml

```diff
@@ -0,0 +1,43 @@
+name: Neurodamus Wheels
+
+on:
+  push:
+    tags:
+      - '*'
+  workflow_dispatch:
```

Comment:
Shall we set different rules for build, test and upload?
We build and test the distribution files when having pull requests to `main`
We build/test/upload when pushing to `tags` 

In this way, each change in the PR is tested for building the wheel. 

That's what we do currently in the GitLab CI although we upload to the internal `devpi`.

Reply:
Sorry, I may have not described my request well. I would prefer to build, test and push in the same workflow, different steps. If we do the pushing in a different workflow, we would have to check whether the wheel is built before pushing.

I think you can set the rule to activate the push `step` using `if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')`like in the [libsonata workflow](https://github.com/BlueBrain/libsonata/blob/0f7b4e4232c76d118de64dfbfe948ecc7d1c7780/.github/workflows/publish-sdist-wheels.yml) .